code/src/reducers/tasks.js

```diff
@@ -0,0 +1,50 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const taskData = [
+  { id: 1, text: 'Book hotel in Ubud', complete: false },
+  { id: 2, text: 'Add projects in my portfolio', complete: false },
+  { id: 3, text: 'Make a pitch about myself', complete: false }
+];
+
+const initialState = {
+  taskData,
+  currentTaskIndex: 0
+};
+
+export const tasks = createSlice({
+  name: 'tasks',
+  initialState,
+  reducers: {
+    addTodo: (state, action) => {
+      console.log(action);
+      state.taskData.push({ id: Date.now(), text: action.payload });
+    },
+    toggleCompleteTask: (state, action) => {
+      const chooseTask = state.taskData.find(txt => txt.id === action.payload);
+
+      if (chooseTask) {
+        chooseTask.complete = !chooseTask.complete;
+      }
+      if (chooseTask) {
```

Comment:
Since this has the same condition as the statement above, the two could be merged together:

```js
if (chooseTask) {
  chooseTask.complete = !chooseTask.complete;
  chooseTask.currentTaskIndex += 1;
}
```

But also, looking at your state, it looks like this `currentTaskIndex` index thing isn't being used. If you were to use it, it looks like you want to do `state.currentTaskIndex += 1` as `chooseTask.currentTaskIndex += 1` will add it to the task object which your toggling completion of in this action.